### PR TITLE
Upgraded maestro to 3.0.0 and simplified the recipe

### PR DESCRIPTION
**Pull request by whygoyal**
1. Removed greaselib
1. Removed git patches as greaselib is no longer required to compile maestro and
event FD2 patch is upstreamed
1. Removed the init scripts as we now use systemd
1. Removed unused patches for all targets
1. Using the same maestro configuration for all the targets 
1. Added new yocto env variables to edge bbclass

This compiles successfully. Integration test not yet performed.

### CODE REVIEW

**Comment by JanneKiiskila** [comment]
@whygoyal  - why do we even have the target specific files for maestro? I've been going through the files and they all seem to be identical. What in maestro is dependent on the target HW?

File: recipes-wigwag/maestro/maestro_0.0.2.inc [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+SRCREV = "v3.0.0"
+
+SYSTEMD_PACKAGES = "${PN}"
+SYSTEMD_SERVICE_${PN} = "maestro.service \
+maestro-watcher.path \
+maestro-watcher.service"
+
+SYSTEMD_AUTO_ENABLE_${PN} = "enable"
+
+RUN_CONFIG_DIR="wigwag/etc/run"
```
Comment (petedyerarm) [inline recipes-wigwag/maestro/maestro_0.0.2.inc:33]: Can we replace the /wigwag style folder names to an environment variable?

**Review by petedyerarm — CHANGES_REQUESTED** [review]

**Review by kivaisan — APPROVED** [review]

File: recipes-wigwag/maestro/maestro_0.0.2.inc (hunk 1 above)
Comment (whygoyal) [reply recipes-wigwag/maestro/maestro_0.0.2.inc]: @petedyerarm Added new variables to edge bbclass.

**Review by petedyerarm — APPROVED** [review]
LGTM

File: recipes-wigwag/maestro/maestro_0.0.2.inc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+inherit pkgconfig systemd go gitpkgv edge
+
+RT_SERVICE_FILE = "maestro.service"
+PR = "r0"
+
+SRC_URI = "git://${GO_IMPORT};protocol=https \
+file://${RT_SERVICE_FILE} \
+file://maestro-watcher.service \
+file://maestro-watcher.path \
+file://maestro-config-uz.yaml \
```
Comment (petedyerarm) [inline recipes-wigwag/maestro/maestro_0.0.2.inc:22]: I think maestro-config-uz.yaml has been deleted. Possilby replaced by maestro-config.yaml

**Review by petedyerarm — CHANGES_REQUESTED** [review]

File: recipes-wigwag/maestro/maestro_0.0.2.inc (hunk 2 above)
Comment (whygoyal) [reply recipes-wigwag/maestro/maestro_0.0.2.inc]: Good catch @petedyerarm. I have fixed it.

**Comment by petedyerarm** [comment]
The CI build fails:

[2021-11-15T19:32:01.784Z] Parsing recipes...ERROR: ParseError at /jenkins_ssd/workspace/ws/repo-meta-pelion-edge_PR-357/build-lmp/conf/../../layers/meta-pelion-edge/recipes-wigwag/maestro/maestro_0.0.2.inc:35: unparsed line: 'RUN_CONFIG_DIR=${EDGE_RUN}'

File: recipes-wigwag/maestro/maestro_0.0.2.inc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+SYSTEMD_PACKAGES = "${PN}"
+SYSTEMD_SERVICE_${PN} = "maestro.service \
+maestro-watcher.path \
+maestro-watcher.service"
+
+SYSTEMD_AUTO_ENABLE_${PN} = "enable"
+
+RUN_CONFIG_DIR=${EDGE_CONFIG}
+TEMPLATE_CONFIG_DIR=${EDGE_TEMPLATE}
+MAESTRO_JOBS_LOG_DIR=${EDGE_LOG}
```
Comment (petedyerarm) [inline recipes-wigwag/maestro/maestro_0.0.2.inc:37]:
```suggestion
RUN_CONFIG_DIR="${EDGE_CONFIG}"
TEMPLATE_CONFIG_DIR="${EDGE_TEMPLATE}"
MAESTRO_JOBS_LOG_DIR="${EDGE_LOG}"
```

File: recipes-wigwag/maestro/maestro_0.0.2.inc [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+FILESEXTRAPATHS_prepend := "${THISDIR}/maestro:"
+FILES_${PN} += "\
+    ${EDGE_BIN}/*\
+    ${EDGE_LIB}/*\
+    ${EDGE_BIN}/maestro \
+    /${RUN_CONFIG_DIR}/*\
+    /${TEMPLATE_CONFIG_DIR}/*\
+    /${MAESTRO_JOBS_LOG_DIR}\
```
Comment (petedyerarm) [inline recipes-wigwag/maestro/maestro_0.0.2.inc:46]:
```suggestion
    ${RUN_CONFIG_DIR}/*\
    ${TEMPLATE_CONFIG_DIR}/*\
    ${MAESTRO_JOBS_LOG_DIR}\
```

**Review by petedyerarm — CHANGES_REQUESTED** [review]

File: classes/edge.bbclass [hunk 5]
```diff
@@ -46,6 +46,9 @@ EDGE_ETC = "/wigwag/system/etc"
 EDGE_DATA = "/userdata"
 EDGE_STATE = "/wigwag/system/var/lib"
 EDGE_OPT = "/wigwag/system/opt"
+EDGE_CONFIG = "/wigwag/etc/run"
```
Comment (petedyerarm) [inline classes/edge.bbclass:49]: At least EDGE_CONFIG needs to be added to the edge_replace_vars() function below. 

Maybe EDGE_TEMPLATE and EDGE_LOG as well

**Review by petedyerarm — CHANGES_REQUESTED** [review]

**Comment by whygoyal** [comment]
Thanks @petedyerarm for the comments. I have fixed the issues and have manually validated the PR. 
This is now ready to be merged.

**Review by petedyerarm — APPROVED** [review]
LGTM

**Review by kivaisan — APPROVED** [review]

**Pull request merged**